Plugins/Public/docking_modules/Utilities.cpp

```diff
@@ -0,0 +1,306 @@
+		}
+
+	hbegin:
+		while (getline(file, line))
+		{
+			for (string &del : hookExtLinesToDelete)
+				if (boost::algorithm::starts_with(line, del))
+					goto hbegin;
+
+			newFile << line << endl;
+		}
+
+		file.close();
+
+		// Add hookext data.
+		for (string &add : hookExtLinesToAdd)
+			newFile << add << endl;
+
+		newFile.close();
+		remove(path.c_str());
+		rename(tempPath.c_str(), path.c_str());
+	}
+}
+
+// My own function to read specific values from FL save file without ini headers. Gets them in raw string format. Returns result of check if file exists.
+void ReadFLFile(map<string, vector<string>> &variables, string &path)
+{
+	ifstream file(path);
+	if (file.is_open())
+	{
+		string line;
+		while (getline(file, line))
+		{
+			for (map<string, vector<string>>::iterator it = variables.begin(); it != variables.end(); ++it)
+			{
+				if (boost::algorithm::starts_with(line, it->first))
+				{
+					uint pos = it->first.size();
+
+					// Reject partial matches.
+					if (line[pos] == ' ')
+					{
+						// pos + 3 - we skip whitespace, equation symbol and whitespace.
+						it->second.push_back(line.substr(pos + 3));
+						break;
+					}
+				}
+			}
+		}
+
+		file.close();
+	}
+}
+
+// Expanded version of previous function.
+void ReadFLFile(map<string, vector<string>> &variables, map<string, string> &hookExtData, string &path)
+{
+	ifstream file(path);
+	if (file.is_open())
+	{
+		string line;
+		while (getline(file, line))
+		{
+			if (boost::algorithm::starts_with(line, "[flhook]"))
+				goto HookExtCheck;
+
+			for (map<string, vector<string>>::iterator it = variables.begin(); it != variables.end(); ++it)
+			{
+				if (boost::algorithm::starts_with(line, it->first))
+				{
+					uint pos = it->first.size();
+
+					// Reject partial matches.
+					if (line[pos] == ' ')
+					{
+						// pos + 3 - we skip whitespace, equation symbol and whitespace.
+						it->second.push_back(line.substr(pos + 3));
+						break;
+					}
+				}
+			}
+		}
+
+		// No lines to read remain, avoid HookExtCheck and go directly to exit procedure.
+		goto Exit;
+
+	HookExtCheck:
+		while (getline(file, line))
+		{
+			uint pos = line.find(" ");
+			hookExtData.insert(make_pair(line.substr(0, pos), line.substr(pos + 3)));
+		}
+
+	Exit:
+		file.close();
+	}
+}
+
+// Converts text in UCS-2 FL encoding to hex string.
+string DecodeWStringToStringOfBytes(wstring &wstr)
+{
+	string bytestr;
+	bytestr.reserve(wstr.size() * 4);
+
+	for (wchar_t c : wstr)
+	{
+		char buf[4];
+		sprintf(buf, "%04x", c);
+		bytestr += buf;
+	}
+
+	return bytestr;
+}
+
+// Converts hex string to UCS-2 text back.
+wstring EncodeWStringFromStringOfBytes(string &bytestr)
+{
+	wstring charName;
+
+	for (uint i = 0; i < bytestr.size() / 4; ++i)
+	{
+		charName.push_back((wchar_t)strtol(bytestr.substr(i * 4, 4).c_str(), NULL, 16));
+	}
+
+	return charName;
+}
+
+// Get vector of strings from one string by separating it by some char.
+vector<string> GetParams(string &str, char splitChar)

```

Comment:
this looks like a simple explode/split function, surely boost has something for it

Reply:
It is more convenient than boost one.